packages/pwa-kit-react-sdk/src/ssr/browser/main.jsx

```diff
@@ -73,18 +73,22 @@ export const start = () => {
         .then(() => new Promise((resolve) => loadableReady(resolve)))
         .then(() => {
             ReactDOM.hydrate(
-                <Router>
-                    <DeviceContext.Provider value={{type: window.__DEVICE_TYPE__}}>
-                        <AppConfig locals={locals}>
-                            <Switch
-                                error={error}
-                                appState={window.__PRELOADED_STATE__}
-                                routes={routes}
-                                App={WrappedApp}
-                            />
-                        </AppConfig>
-                    </DeviceContext.Provider>
-                </Router>,
+                <PageRequestContext.Provider value={undefined}>
+                    <PageResponseContext.Provider value={undefined}>
```

Comment:
Similar to when used within `getProps`, the `Request` and `Response` objects are not available on the client-side. 

However, unlike the hooks version, `getProps` allows you to serialize data, which is then available later on the client-side. This fundamental difference introduces a caveat that properties of Request/Response should not be used for rendering a component. See the example in template-typescript-minimal for more details.